### PR TITLE
server/embed: address golangci var-naming issues

### PR DESCRIPTION
**Pull request by ivanvc**
Addresses issues in ListenPeerUrls, ListenClientUrls, ListenClientHttpUrls, AdvertisePeerUrls, AdvertiseClientUrls.

Following the same approach as in #17674, a linter ignore is added to these exported fields. However, non-exported functions and exported struct fields used only for JSON marshaling can be safely renamed.

Part of #17578.

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: server/embed/config.go [hunk 1]
```diff
@@ -449,11 +452,11 @@ type configYAML struct {
...
-	ListenPeerUrls       string `json:"listen-peer-urls"`
-	ListenClientUrls     string `json:"listen-client-urls"`
-	ListenClientHttpUrls string `json:"listen-client-http-urls"`
-	AdvertisePeerUrls    string `json:"initial-advertise-peer-urls"`
-	AdvertiseClientUrls  string `json:"advertise-client-urls"`
+	ListenPeerURLs       string `json:"listen-peer-urls"`
+	ListenClientURLs     string `json:"listen-client-urls"`
+	ListenClientHTTPURLs string `json:"listen-client-http-urls"`
+	AdvertisePeerURLs    string `json:"initial-advertise-peer-urls"`
+	AdvertiseClientURLs  string `json:"advertise-client-urls"`
```
Comment (ivanvc) [inline server/embed/config.go:459]: As stated in the PR description, these are exported fields from an unexported struct. It's internally used to marshal/unmarshal JSONs. So, it should be fine to rename.

File: server/etcdmain/config_test.go [hunk 2]
```diff
@@ -61,10 +61,10 @@ func TestConfigFileMemberFields(t *testing.T) {
...
 		SnapshotCount          uint64 `json:"snapshot-count"`
 		SnapshotCatchUpEntries uint64 `json:"experimental-snapshot-catch-up-entries"`
-		ListenPeerUrls         string `json:"listen-peer-urls"`
-		ListenClientUrls       string `json:"listen-client-urls"`
-		ListenClientHttpUrls   string `json:"listen-client-http-urls"`
-		AdvertiseClientUrls    string `json:"advertise-client-urls"`
+		ListenPeerURLs         string `json:"listen-peer-urls"`
+		ListenClientURLs       string `json:"listen-client-urls"`
+		ListenClientHTTPURLs   string `json:"listen-client-http-urls"`
+		AdvertiseClientURLs    string `json:"advertise-client-urls"`
```
Comment (ivanvc) [inline server/etcdmain/config_test.go:67]: As stated in the PR description, these are used only for tests marshaling the config into JSON, so it should be safe to rename them.

File: server/embed/config.go (hunk 1 above)
Comment (serathius) [reply server/embed/config.go:459]: Right

**Review by serathius — APPROVED** [review]

**Review by ahrtr — APPROVED** [review]

**Pull request merged**